packages/aws-cdk-lib/aws-events/test/event-bus.test.ts

```diff
@@ -625,4 +626,21 @@ describe('event bus', () => {
       actions: ['events:PutEvents'],
     }))).toThrow('Event Bus policy statements must have a sid');
   });
+
+  test('Event Bus with a customer managed key', () => {
+    // GIVEN
+    const app = new App();
+    const stack = new Stack(app, 'Stack');
+    const key = new kms.Key(stack, 'Key');
+
+    // WHEN
+    new EventBus(stack, 'Bus', {
+      kmsKey: key,
+    });
+
+    // THEN
+    Template.fromStack(stack).hasResourceProperties('AWS::Events::EventBus', {
+      KmsKeyIdentifier: stack.resolve(key.keyArn),
+    });
```

Comment:
Can you please also add an assertion on the key policy?
```ts
Template.fromStack(stack).hasResourceProperties('AWS::KMS::Key', {
          KeyPolicy: {
            'Statement': [ ... ]
          }
})
```

Reply:
Thanks.
I added an assertion for the key policy.